src/main/java/com/google/sps/data/OrganizationUpdater.java

```diff
@@ -130,10 +135,17 @@ private void setOrganizationProperty(String propertyKey, String formValue) {
   private ArrayList<String> translateEmailsToIds(ArrayList<String> emails) {
     ArrayList<String> userIds = new ArrayList<String>();
     
-    // TODO(): Implement correct way of getting user by email once that is decided on
     for(String email : emails) {
       GivrUser newUser = GivrUser.getUserByEmail(email);
-      userIds.add(newUser.getUserId());
+      String userId = newUser.getUserId();
+      // If failed will return "" (?) and user will be added to the invited moderator property 
+      // and not the offical list
+      if(userId.equals("")) {
+        Set<String> invitedModerators = new HashSet((HashSet)this.entity.getProperty("invitedModerators"));
+        invitedModerators.add(email);
+      } else {
+        userIds.add(newUser.getUserId());
+      }
```

Comment:
Per other comment -- take care to separate out the mutation functions from the read functions, and to make sure what's happening is obvious from just the name of the function. Eg reading "translateEmailsToIds": I would not guess that the emails are added to the moderator list.

Can you decompose this into a set of blocks that have obvious names and no side effects?

Ex to add a moderator:
array emails[]
for (email in emails) 
  user = UserFromEmail
  org.addmoderator(user)

Reply:
I refactored this code and added a few comments!